mr/src/main/java/org/elasticsearch/hadoop/serialization/ScrollReader.java

```diff
@@ -981,7 +981,9 @@ protected Object singletonList(String fieldMapping, Object value, Parser parser)
         Object array = reader.createArray(mapping(fieldMapping, parser));
         // create only one element since with fields, we always get arrays which create unneeded allocations
         List<Object> content = new ArrayList<Object>(1);
-        content.add(value);
+        if (null != value) {
+            content.add(value);
```

Comment:
This is still going to create an empty array for a field that is listed as null. Is this the outcome you'd always expect to see instead of just setting the field to null?

Personally, I can see an array field set to null as equally meaning no array at all. I think we need to hammer down what the correct default should be. Either an empty array or a null field, but I agree that it should not be an array with a null value.